Makefile

```diff
@@ -0,0 +1,29 @@
+# Makefile
+THIS_FILE := $(lastword $(MAKEFILE_LIST))
+include common.mk
+SHELL := /bin/bash
+
+RUNTEST=python -m unittest -v -b
+
+ALLMODULES=$(patsubst %.py, %, $(wildcard test_*.py))
+
+all:
+	@echo "running tests..."
+	${RUNTEST} ${ALLMODULES}
+
+% : test_%.py
+	${RUNTEST} test_$@
+
+venv: venv/bin/activate
+
+venv/bin/activate: requirements.txt
+	test -d .venvbag || virtualenv --python=python3.6 .venvbag
+	.venvbag/bin/pip install -Ur requirements.txt
+	touch .venvbag/bin/activate
+
+deploy: venv
+	@$(MAKE) -f $(THIS_FILE)  all
```

Comment:
Could you just make `all` a dependency?

Reply:
the target `deploy` depends on both, `venv` and `all`. But I my understanding it that you can only make it dependent on one other target without using the `$(MAKE)` construct (that's how I used it for `venv`. but because there are two dependencies I had use `$(MAKE)` here.